dbt-adapters/src/dbt/adapters/catalogs/_integration.py

```diff
@@ -0,0 +1,71 @@
+        table_format (Optional[str]): the table format this catalog uses
+            - this is commonly unique to each catalog type, and should only be required from the user for catalogs that support multiple formats
+        external_volume (Optional[str]): external storage volume identifier
+            - while this is a separate concept from catalogs, we feel it is more user-friendly to group it with the catalog configuration
+            - a result of this grouping is that there can only be one external volume per catalog integration, but many catalogs can share the same volume
+        adapter_properties (Optional[Dict[str, Any]]):
+            - additional, adapter-specific properties are nested here to avoid future collision when expanding the catalog integration protocol
+    """
+
+    name: str
+    catalog_type: str
+    table_format: Optional[str]
+    external_volume: Optional[str]
+    adapter_properties: Optional[Dict[str, Any]]
+
+
+class CatalogIntegration:
+    """
+    Represent a catalog integration for a given user config
+
+    This class should be subclassed by specific catalog integration types in an adapter.
+    A catalog integration is a specific platform's way of interacting with a specific catalog.
+
+    Attributes:
+        name (str): the name of the catalog integration in the data platform, e.g. "my_favorite_iceberg_catalog"
+            - this is required for dbt to correctly reference catalogs by name from model configuration
+            - expected to be unique within the adapter, if not the entire data platform
+        catalog_type (str): the type of the catalog integration in the data platform, e.g. "iceberg_rest"
+            - this is required for dbt to determine the correct method for parsing user configuration
+            - usually a combination of the catalog and the way in which the data platform interacts with it
+        allows_writes (bool): identifies whether this catalog integration supports writes
+            - this is required for dbt to correctly identify whether a catalog is writable during parse time
+            - this is determined by the catalog integration type, hence it is a class attribute
+        table_format (Optional[str]): the table format this catalog uses
+            - this is commonly determined by the catalog integration type, hence it is usually a class attribute
+            - it should only be required from the user for catalogs that support multiple formats
+        external_volume (Optional[str]): external storage volume identifier
+            - while this is a separate concept from catalogs, we feel it is more user-friendly to group it with the catalog configuration
+            - a result of this grouping is that there can only be one external volume per catalog integration, but many catalogs can share the same volume
+    """
+
+    allows_writes: bool = False
```

Comment:
Can we change this to an enum for future-proofing?

Reply:
I'm having a difficult time coming up with an alternative mechanism. What would the values be? Do we want to roll reads/writes into one attribute? Is there another state I'm not thinking of?

Reply:
```
from enum import Enum

class CatalogIntegrationMode(Enum):
    READ = "r"
    WRITE = "w"
```